### PR TITLE
Revise attachments while preserving older versions

### PR DESCRIPTION
**Pull request by miggol**
If you edit an attachment that is already attached to something else, the portal will instead seamlessly save a copy so older proposals don't change.

To test:

Revise an application with an Attachment attached. Modify the attachment in the revision, and note that it gets a new pk while keeping all attributes that you would expect. The old application and its attachment will remain unchanged.

If this looks simple know that I played around with this a lot before getting it right, and the whole rest of the attachments system is pretty much built around this part being easy. Still proud of it, though!

### CODE REVIEW

File: proposals/views/attachment_views.py [hunk 1]
```diff
@@ -39,11 +39,32 @@ def __init__(self, kind=None, other_object=None, extra=False, **kwargs):
...
+            # The following means this instance will get saved under a
+            # new pk, effectively creating a copy.
+            self.instance.pk = None
+            self.instance.id = None
+            self.instance._state.adding = True
+            self.instance.save()
+            # Retrieve and fix the old attachment.
+            instance_manager = type(self.instance).objects
+            old_attachment = instance_manager.get(pk=old_pk)
+            old_attachment.attached_to.remove(self.other_object)
```
Comment (EdoStorm96) [inline proposals/views/attachment_views.py:62]: I was kindoff confused about old_attachment not getting saved here ... But it seems to work as expected, as in, the new object gets removed from the old attachment. I've tested this in the shell.

**Review by EdoStorm96 — CHANGES_REQUESTED** [review]
So besides the little creation bug, this works well AFAICT. Nice and simple. Good job!

File: proposals/views/attachment_views.py (hunk 1 above)
Comment (miggol) [reply proposals/views/attachment_views.py]: I think elsewhere we do save in these cases "for good measure", but I think that we shouldn't. M2m relationships get updated instantly and it's good to be reminded of that now and again. I also sometimes forget if it's necessary to save them, but it isn't.

**Comment by miggol** [comment]
I've fixed the creation bug, thanks for pointing that out. I've made the solution quite a bit more verbose to make more clear what's going on.

**Review by EdoStorm96 — APPROVED** [review]
Haven't tested ... But looks good!

**Pull request merged**